source/fem-dem/ib_particles_dem.cc

```diff
@@ -606,105 +606,221 @@ IBParticlesDEM<dim>::integrate_particles_motion(const double dt,
-      lubrication_wall_force.resize(dem_particles.size());
-      lubrication_wall_torque.clear();
-      lubrication_wall_torque.resize(dem_particles.size());
-
-      // Calculate particle-particle and particle-wall contact force
-      calculate_pp_contact_force(dt_dem, contact_force, contact_torque);
-      calculate_pw_contact_force(dt_dem,
-                                 contact_wall_force,
-                                 contact_wall_torque);
-      calculate_pp_lubrication_force(
-        dt_dem, h_max, h_min, mu, lubrication_force, lubrication_torque);
-      calculate_pw_lubrication_force(dt_dem,
-                                     h_max,
-                                     h_min,
-                                     mu,
-                                     lubrication_wall_force,
-                                     lubrication_wall_torque);
-
-
-      for (unsigned int p_i = 0; p_i < dem_particles.size(); ++p_i)
+      // Initialize vector to contain the RK 4 step
+      std::vector<Tensor<1, 3>> last_velocity;
+      std::vector<Point<dim>>   last_position;
+      std::vector<Tensor<1, 3>> last_omega;
+      last_velocity = std::vector<Tensor<1, 3>>(dem_particles.size());
+      last_position = std::vector<Point<dim>>(dem_particles.size());
+      last_omega    = std::vector<Tensor<1, 3>>(dem_particles.size());
+
+      std::vector<std::vector<Tensor<1, 3>>>   k_velocity;
+      std::vector<std::vector<Tensor<1, dim>>> k_position;
+      std::vector<std::vector<Tensor<1, 3>>>   k_omega;
+      std::vector<std::vector<Tensor<1, 3>>>   k_impulsion;
+      std::vector<std::vector<Tensor<1, 3>>>   k_omega_impulsion;
+      std::vector<std::vector<Tensor<1, 3>>>   k_contact_impulsion;
+      std::vector<std::vector<Tensor<1, 3>>>   k_omega_contact_impulsion;
+
+      k_velocity =
+        std::vector<std::vector<Tensor<1, 3>>>(dem_particles.size(),
+                                               std::vector<Tensor<1, 3>>(4));
+      k_position = std::vector<std::vector<Tensor<1, dim>>>(
+        dem_particles.size(), std::vector<Tensor<1, dim>>(4));
+      k_omega =
+        std::vector<std::vector<Tensor<1, 3>>>(dem_particles.size(),
+                                               std::vector<Tensor<1, 3>>(4));
+      k_impulsion =
+        std::vector<std::vector<Tensor<1, 3>>>(dem_particles.size(),
+                                               std::vector<Tensor<1, 3>>(4));
+      k_omega_impulsion =
+        std::vector<std::vector<Tensor<1, 3>>>(dem_particles.size(),
+                                               std::vector<Tensor<1, 3>>(4));
+      k_contact_impulsion =
+        std::vector<std::vector<Tensor<1, 3>>>(dem_particles.size(),
+                                               std::vector<Tensor<1, 3>>(4));
+      k_omega_contact_impulsion =
+        std::vector<std::vector<Tensor<1, 3>>>(dem_particles.size(),
+                                               std::vector<Tensor<1, 3>>(4));
```

Comment:
Why not allocate them at the same line you declare them? It would be a bit cleaner no?